src/storage/FileResourceStore.ts

```diff
@@ -243,28 +231,27 @@ export class FileResourceStore implements ResourceStore {
     if (representation.dataType !== DATA_TYPE_BINARY) {
       throw new UnsupportedMediaTypeHttpError('FileResourceStore only supports binary representations.');
     }
-    const _path = this.parseIdentifier(identifier);
-    const { linkType, raw } = representation.metadata;
+    const [ , path, slug ] = /(?<path>.*\/)(?<slug>[^/]*\/?)/u.exec(this.parseIdentifier(identifier)) ?? [];
```

Comment:
Would add `^` at the beginning of the regex and `$` at the end to make clear that the entire string needs to be matched.

Since you're just extracting the results from the output array, there is no need to name the matched parts, so that can be dropped.

In the case the incoming URL ends with a slash slug is going to be empty. You can fix this by using a `+` instead of a `*` in the slug part.

So full regex would then be `/^(.*\/)([^/]+\/?)$/u`

Reply:
I've added the `^` and `$` but the group naming is required by the linter.
(`ESLint: Capture group '(.*\/)' should be converted to a named or non-capturing group.(prefer-named-capture-group)`)
The latter is also a good one, but if the path is then only `/`, both are undefined. However, this would then be the root container, and for this we might not want to allow it to be modified after all.

Reply:
Ah didn't know that the linter required it.

Yes this regex would not work for root, but since that is by default always a container I don't think a PUT is allowed there?

Reply:
Why are the parentheses there in the first place?

Reply:
Let's disable that linter rule, it doesn't make sense.

By using `/^(.*\/)([^/]+\/?)?$/u` (extra question mark), it also works for root?

Reply:
Make that `/\/([^/]+\/?)?$/u`, no need to match `.*` in the beginning.

Reply:
Yes, that works for root, but I'll add an extra check to throw an error if it is performed on root too.

Should I adjust the linter? (where?)